src/process.c

```diff
@@ -65,12 +67,53 @@ struct process * process_create( unsigned code_size, unsigned stack_size )
 
 	p->kstack = memory_alloc_page(1);
 	p->entry = PROCESS_ENTRY_POINT;
+	memset(p->fd_table, 0, sizeof(p->fd_table));
 
 	process_stack_init(p);
 
 	return p;
 }
 
+static int get_available_fd()
+{
+	uint32_t i;
+	for (i = 0; i < PROCESS_FD_COUNT; i++)
+	{
+		if (current->fd_table[i] == 0)
+			return i;
+	}
+	return -1;
+}
+
+int process_open_file(const char *filename, int8_t mode)
+{
+	int fd = get_available_fd();
+	int inode_number = file_memory_inode_index(filename);
```

Comment:
A big picture issue we need to think about here is the namespace of our filesystem.
Is it like Unix, in which devices appear in /dev and filesystems are grafted into a tree?

Like VMS, in which devices have a unique namespace? (CDROM:/path/to/file)

Like DOS in which devices have abstract names?  (C:/path/to/file

Reply:
Ideally, I would prefers something like this:

- Devices are their own registry that you would view and interact with through commands like `ls-dev`, `write <device> < data`, and others
- Devices could implement `fs` apis to indicate that they are capable of being read/written/mounted/etc as a filesystem.
- You would mount a filesystem with `mount sda1 c` and access files like this `c:/dir/file.txt`

So it'd be namespacing like DOS or VMS, but only with devices that are clearly a filesystem.

However, all of this might be moot if users prefer something that feels like linux.